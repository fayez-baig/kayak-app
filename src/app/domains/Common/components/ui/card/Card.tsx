import { FC, ReactElement } from "react";
import { CardProps } from "./types";
import "./card.scss";

const Card: FC<CardProps> = ({
  logoUrl,
  title,
  alliance,
  phone,
  website,
}): ReactElement => {
  return (
    <div className="card-wrapper">
      <img
        className="card-logo"
        src={`https://kayak.com${logoUrl}`}
        alt={title}
      />

      <div className="card-container">
        <div className="card-title">{title}</div>
        <div className="card-info">
          <div className="card-heading">{alliance}</div>
          <div className="card-heading">{phone}</div>
          <div className="card-subheading">{website}</div>
        </div>
      </div>
    </div>
  );
};

export default Card;

//
