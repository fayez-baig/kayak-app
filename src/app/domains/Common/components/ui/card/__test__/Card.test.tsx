import { createRenderer } from "react-test-renderer/shallow";
import Card from "../Card";

const renderer = createRenderer();

const props = {
  website: "https://subus.cl/",
  alliance: "none",
  phone: "",
  title: "SuBÃºs",
  logoUrl:
    "/rimg/provider-logos/airlines/v/SUBUS.png?crop=false&width=108&height=92&fallback=default1.png&_v=cf7f153576dad48e814135373e3baf34",
};

describe("<Card />", () => {
  it("should render and match the snapshot", () => {
    renderer.render(<Card {...props} />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
