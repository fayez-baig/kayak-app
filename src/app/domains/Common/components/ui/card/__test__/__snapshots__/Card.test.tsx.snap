// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render and match the snapshot 1`] = `
<div
  className="card-wrapper"
>
  <img
    alt="SuBús"
    className="card-logo"
    src="https://kayak.com/rimg/provider-logos/airlines/v/SUBUS.png?crop=false&width=108&height=92&fallback=default1.png&_v=cf7f153576dad48e814135373e3baf34"
  />
  <div
    className="card-container"
  >
    <div
      className="card-title"
    >
      SuBús
    </div>
    <div
      className="card-info"
    >
      <div
        className="card-heading"
      >
        none
      </div>
      <div
        className="card-heading"
      >
        
      </div>
      <div
        className="card-subheading"
      >
        https://subus.cl/
      </div>
    </div>
  </div>
</div>
`;
